packages/htmljs/visitors.js

```diff
@@ -121,76 +126,73 @@ TransformingVisitor.def({
+    if (obj.textMode != null) {
       return obj;
     }
     if ('content' in obj) {
       obj.content = this.visit(obj.content, ...args);
     }
-    if ('elseContent' in obj){
+    if ('elseContent' in obj) {
       obj.elseContent = this.visit(obj.elseContent, ...args);
     }
     return obj;
   },
   visitFunction: IDENTITY,
-  visitTag: function (tag, ...args) {
-    var oldChildren = tag.children;
-    var newChildren = this.visitChildren(oldChildren, ...args);
+  visitTag(tag, ...args) {
+    const oldChildren = tag.children;
+    const newChildren = this.visitChildren(oldChildren, ...args);
 
-    var oldAttrs = tag.attrs;
-    var newAttrs = this.visitAttributes(oldAttrs, ...args);
+    const oldAttrs = tag.attrs;
+    const newAttrs = this.visitAttributes(oldAttrs, ...args);
 
-    if (newAttrs === oldAttrs && newChildren === oldChildren)
-      return tag;
+    if (newAttrs === oldAttrs && newChildren === oldChildren) return tag;
 
-    var newTag = getTag(tag.tagName).apply(null, newChildren);
+    const newTag = getTag(tag.tagName).apply(null, newChildren);
     newTag.attrs = newAttrs;
     return newTag;
   },
-  visitChildren: function (children, ...args) {
+  visitChildren(children, ...args) {
     return this.visitArray(children, ...args);
   },
   // Transform the `.attrs` property of a tag, which may be a dictionary,
   // an array, or in some uses, a foreign object (such as
   // a template tag).
-  visitAttributes: function (attrs, ...args) {
+  visitAttributes(attrs, ...args) {
     if (isArray(attrs)) {
-      var result = attrs;
-      for (var i = 0; i < attrs.length; i++) {
-        var oldItem = attrs[i];
-        var newItem = this.visitAttributes(oldItem, ...args);
+      let result = attrs;
+      for (let i = 0; i < attrs.length; i++) {
+        const oldItem = attrs[i];
+        const newItem = this.visitAttributes(oldItem, ...args);
         if (newItem !== oldItem) {
           // copy on write
-          if (result === attrs)
-            result = attrs.slice();
+          if (result === attrs) result = attrs.slice();
           result[i] = newItem;
         }
       }
       return result;
     }
 
     if (attrs && isConstructedObject(attrs)) {
-      throw new Error("The basic TransformingVisitor does not support " +
-                      "foreign objects in attributes.  Define a custom " +
-                      "visitAttributes for this case.");
+      throw new Error('The basic TransformingVisitor does not support ' +
+        'foreign objects in attributes.  Define a custom ' +
+        'visitAttributes for this case.');
     }
 
-    var oldAttrs = attrs;
-    var newAttrs = oldAttrs;
+    const oldAttrs = attrs;
+    let newAttrs = oldAttrs;
     if (oldAttrs) {
-      var attrArgs = [null, null];
+      const attrArgs = [null, null];
       attrArgs.push.apply(attrArgs, arguments);
-      for (var k in oldAttrs) {
-        var oldValue = oldAttrs[k];
+      for (const k in oldAttrs) {
```

Comment:
Use `Object.getOwnPropertyNames` instead of `for..in`

```
      Object.getOwnPropertyNames(oldAttrs).forEach(k => {
        const oldValue = oldAttrs[k];
        attrArgs[0] = k;
        attrArgs[1] = oldValue;
        const newValue = this.visitAttribute.apply(this, attrArgs);
        if (newValue !== oldValue) {
          // copy on write
          if (newAttrs === oldAttrs) newAttrs = _assign({}, oldAttrs);
          newAttrs[k] = newValue;
        }
      });
```